src/MessageContainer.tsx

```diff
@@ -0,0 +1,333 @@
+    }
+    return null
+  }
+
+  scrollTo(options: { animated?: boolean; offset: number }) {
+    if (this.flatListRef && this.flatListRef.current && options) {
+      this.flatListRef.current.scrollToOffset(options)
+    }
+  }
+
+  scrollToBottom = () => {
+    this.scrollTo({ offset: 0, animated: true })
+  }
+
+  handleOnScroll = (event: any) => {
+    if (event.nativeEvent.contentOffset.y > this.props.scrollToBottomOffset!) {
+      this.setState({ showScrollBottom: true })
+    } else {
+      this.setState({ showScrollBottom: false })
+    }
+  }
+
+  renderRow = ({ item, index }: ListRenderItemInfo<IMessage>) => {
+    if (!item._id && item._id !== 0) {
+      console.warn(
+        'GiftedChat: `_id` is missing for message',
+        JSON.stringify(item),
+      )
+    }
+    if (!item.user) {
+      if (!item.system) {
+        console.warn(
+          'GiftedChat: `user` is missing for message',
+          JSON.stringify(item),
+        )
+      }
+      item.user = { _id: 0 }
+    }
+    const { messages, user, ...restProps } = this.props
+    if (messages && user) {
+      const previousMessage = messages[index + 1] || {}
+      const nextMessage = messages[index - 1] || {}
+
+      const messageProps: Message['props'] = {
+        ...restProps,
+        key: item._id,
+        currentMessage: item,
+        previousMessage,
+        nextMessage,
+        position: item.user._id === user._id ? 'right' : 'left',
+      }
+
+      if (this.props.renderMessage) {
+        return this.props.renderMessage(messageProps)
+      }
+      return <Message {...messageProps} />
+    }
+    return null
+  }
+
+  renderHeaderWrapper = () => (
+    <View style={styles.headerWrapper}>{this.renderLoadEarlier()}</View>
+  )
+
+  renderScrollToBottomWrapper() {
+    const scrollToBottomComponent = (
+      <View style={styles.scrollToBottomStyle}>
+        <TouchableOpacity
+          onPress={this.scrollToBottom}
+          hitSlop={{ top: 5, left: 5, right: 5, bottom: 5 }}
+        >
+          <Text>V</Text>
+        </TouchableOpacity>
+      </View>
+    )
+
+    if (this.props.scrollToBottomComponent) {
+      return (
+        <TouchableOpacity
+          onPress={this.scrollToBottom}
+          hitSlop={{ top: 5, left: 5, right: 5, bottom: 5 }}
+        >
+          {this.props.scrollToBottomComponent}
+        </TouchableOpacity>
+      )
+    }
+    return scrollToBottomComponent
+  }
+
+  keyExtractor = (item: IMessage) => `${item._id}`
+
+  render() {
+    if (
+      !this.props.messages ||
+      (this.props.messages && this.props.messages.length === 0)
+    ) {
+      return <View style={styles.container} />
+    }
+    return (
+      <View
+        style={
+          this.props.alignTop ? styles.containerAlignTop : styles.container
+        }
+      >
+        {this.state.showScrollBottom && this.props.scrollToBottom
+          ? this.renderScrollToBottomWrapper()
+          : null}
+        <FlatList
+          ref={this.flatListRef}
```

Comment:
Here is an error. This means that when sending a new message, do not scroll down. I am right?

Reply:
I you have time make PR, it will be more explicit. Thanks